File: 004/solve.py
#!usr/bin/python

def palindrome(x):
    result = 0
    while x > 0:
        result = result*10+x%10
        x /= 10
    return result

result = 0
for i in xrange(1, 1000):
    for j in xrange(i, 1000):
        ij = i*j
        if ij == palindrome(ij) and ij > result:
            result = ij

print result

File: 004/solve.go
package main

import "fmt"

func polindrome(i int) int {
	result := 0

	for i > 0 {
		result = result*10 + i%10
		i /= 10
	}

	return result
}

func main() {
	result := 0

	for i := 10; i < 1000; i++ {
		for j := 10; j < 1000; j++ {
			ij := i * j
			if ij == polindrome(ij) && ij > result {
				result = ij
			}
		}
	}

	fmt.Println(result)
}
